float64 x
float64 y
float64 z
float64 radius
int64 num_points
---
bool ok
